lib/typing/redshift_test.go

```diff
@@ -3,63 +3,101 @@ package typing
+
 	"github.com/stretchr/testify/assert"
 )
 
 func TestRedshiftTypeToKind(t *testing.T) {
+	type rawTypeAndPrecision struct {
+		rawType   string
+		precision string
+	}
+
 	type _testCase struct {
 		name       string
-		rawTypes   []string
+		rawTypes   []rawTypeAndPrecision
 		expectedKd KindDetails
 	}
 
 	testCases := []_testCase{
 		{
-			name:       "Integer",
-			rawTypes:   []string{"integer", "bigint", "INTEGER"},
+			name: "Integer",
+			rawTypes: []rawTypeAndPrecision{
+				{rawType: "integer"},
+				{rawType: "bigint"},
+				{rawType: "INTEGER"},
+			},
 			expectedKd: Integer,
 		},
 		{
-			name:       "String",
-			rawTypes:   []string{"character varying"},
-			expectedKd: String,
-		},
-		{
-			name:       "String",
-			rawTypes:   []string{"character varying"},
+			name: "String w/o precision",
+			rawTypes: []rawTypeAndPrecision{
+				{rawType: "character varying"},
+				{rawType: "character varying(65535)"},
+				{
+					rawType:   "character varying",
+					precision: "not a number",
+				},
+			},
 			expectedKd: String,
 		},
 		{
-			name:       "String",
-			rawTypes:   []string{"character varying(65535)"},
-			expectedKd: String,
+			name: "String w/ precision",
+			rawTypes: []rawTypeAndPrecision{
+				{
+					rawType:   "character varying",
+					precision: "65535",
+				},
+			},
+			expectedKd: KindDetails{
+				Kind:                         String.Kind,
+				OptionalRedshiftStrPrecision: ptr.ToInt(65535),
+			},
 		},
 		{
-			name:       "Double Precision",
-			rawTypes:   []string{"double precision", "DOUBLE precision"},
+			name: "Double Precision",
+			rawTypes: []rawTypeAndPrecision{
+				{rawType: "double precision"},
+				{rawType: "DOUBLE precision"},
+			},
 			expectedKd: Float,
 		},
 		{
-			name:       "Time",
-			rawTypes:   []string{"timestamp with time zone", "timestamp without time zone", "time without time zone", "date"},
+			name: "Time",
+			rawTypes: []rawTypeAndPrecision{
+				{rawType: "timestamp with time zone"},
+				{rawType: "timestamp without time zone"},
+				{rawType: "time without time zone"},
+				{rawType: "date"},
+			},
 			expectedKd: ETime,
 		},
 		{
-			name:       "Boolean",
-			rawTypes:   []string{"boolean"},
+			name: "Boolean",
+			rawTypes: []rawTypeAndPrecision{
+				{rawType: "boolean"},
+			},
 			expectedKd: Boolean,
 		},
 		{
-			name:       "numeric",
-			rawTypes:   []string{"numeric(5,2)", "numeric(5,5)"},
+			name: "numeric",
+			rawTypes: []rawTypeAndPrecision{
+				{rawType: "numeric(5,2)"},
+				{rawType: "numeric(5,5)"},
+			},
 			expectedKd: EDecimal,
 		},
 	}
 
 	for _, testCase := range testCases {
-		for _, rawType := range testCase.rawTypes {
-			kd := RedshiftTypeToKind(rawType)
+		for _, rawTypeAndPrec := range testCase.rawTypes {
+			kd := RedshiftTypeToKind(rawTypeAndPrec.rawType, rawTypeAndPrec.precision)
 			assert.Equal(t, testCase.expectedKd.Kind, kd.Kind, testCase.name)
+
+			if kd.OptionalRedshiftStrPrecision != nil {
```

Comment:
Can we just do:
`assert.Equal(t, *testCase.expectedKd.OptionalRedshiftStrPrecision, *kd.OptionalRedshiftStrPrecision, testCase.name)`
Without the check that `kd.OptionalRedshiftStrPrecision ` is not `nil`, which will also ensure that we're not failing to set a `kd.OptionalRedshiftStrPrecision` for a test case that expects one.

Reply:
Or I suppose add an `else` with `assert.Nil(t, *kd.OptionalRedshiftStrPrecision, testCase.name)`